--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -489,9 +489,9 @@
     )
 
     IF(WIN32 AND USE_TPM2)
-        TARGET_LINK_LIBRARIES(${LIBS} ${LIBEVENT} ${LIBEVENT_PTHREADS} tbs ncrypt crypt32)
+        TARGET_LINK_LIBRARIES(${LIBDOGECOIN_NAME} PUBLIC ${LIBEVENT} ${LIBEVENT_PTHREADS} tbs ncrypt crypt32)
     ELSE()
-        TARGET_LINK_LIBRARIES(${LIBS} ${LIBEVENT} ${LIBEVENT_PTHREADS})
+        TARGET_LINK_LIBRARIES(${LIBDOGECOIN_NAME} PUBLIC ${LIBEVENT} ${LIBEVENT_PTHREADS})
     ENDIF()
     IF(USE_TESTS)
         TARGET_SOURCES(tests ${visibility}
@@ -512,9 +512,9 @@
     ADD_EXECUTABLE(such src/cli/such.c)
     INSTALL(TARGETS such RUNTIME)
     IF (WIN32 AND USE_TPM2)
-        TARGET_LINK_LIBRARIES(such ${LIBS} tbs ncrypt crypt32)
+        TARGET_LINK_LIBRARIES(such ${LIBDOGECOIN_NAME} tbs ncrypt crypt32)
     ELSE()
-        TARGET_LINK_LIBRARIES(such ${LIBS})
+        TARGET_LINK_LIBRARIES(such ${LIBDOGECOIN_NAME})
     ENDIF()
     TARGET_INCLUDE_DIRECTORIES(such ${visibility} src)
 
